### PR TITLE
infra: switch to using private ECR repo

### PR DESCRIPTION
**Pull request by skipcloud**
The sidecar isn't used anywhere outside of our infra that would justify having the images be published publically.

It'll make our job easier if we have this in a private repo.

Ticket number NEC-2356
https://deliveroo.atlassian.net/browse/NEC-2356

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -131,19 +125,17 @@ jobs:
       - run:
           name: Tag the Docker image
           command: docker tag $(./image) "${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}:staging"
-      - run:
-          name: Push the image to Dockerhub
-          command: docker push "${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}:staging"
+      # - run:
+      #     name: Push the image to Dockerhub
+      #     command: docker push "${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}:staging"
```
Comment (skipcloud) [inline .circleci/config.yml:130]: pushes to dockerhub are failing on staging anyway, but I'm commenting it out while I work on getting the ECR repo push working

File: .circleci/config.yml [hunk 2]
```diff
@@ -119,7 +113,7 @@ jobs:
             - a7:2f:cb:c2:b3:6a:17:c4:8f:3a:8d:77:57:d3:41:bb
       - aws-cli/setup:
           role-arn: ${OIDC_ROLE_ARN}
-          aws-region: us-east-1 # Public ECR Repo
+          aws-region: ECR_AWS_REGION
```
Comment (skipcloud) [inline .circleci/config.yml:116]: requires an env var name not a string

File: infrastructure/modules/basic/versions.tf [hunk 3]
```diff
@@ -1,8 +1,9 @@
 terraform {
   required_providers {
     aws = {
-      source  = "hashicorp/aws"
-      version = "~> 5.70"
+      source                = "hashicorp/aws"
+      version               = "~> 5.70"
+      configuration_aliases = [aws.us_east_1]
```
Comment (skipcloud) [inline infrastructure/modules/basic/versions.tf:6]: adding this to silence a terraform warning, but it'll be removed in the next PR anyway once I completely remove the us east provider

File: infrastructure/modules/basic/locals.tf [hunk 4]
```diff
@@ -1,9 +1,4 @@
 locals {
-  iam_tags_for_shared_policies = {
-    ECRRepoName = local.ecr_repo_name
-  }
-  default_tags = merge(data.aws_default_tags.current.tags, local.iam_tags_for_shared_policies)
-
```
Comment (skipcloud) [inline infrastructure/modules/basic/locals.tf:6]: No longer required. I added this when I was initially working towards using our shared IAM policies, however we needed to pivot due to our specific needs meaning we weren't using our terraform modules to spin up our very simple pipeline

**Review by rajatgupta310198 — APPROVED** [review]

**Pull request merged**